metricbeat/helper/prometheus/textparse.go

```diff
@@ -537,15 +542,19 @@ func ParseMetricFamilies(b []byte, contentType string, ts time.Time) ([]*MetricF
 		var lbls strings.Builder
 		lbls.Grow(len(mets))
 		var labelPairs = []*labels.Label{}
+		var qv string // value of le or quantile label
 		for _, l := range lset.Copy() {
 			if l.Name == labels.MetricName {
 				continue
 			}
 
-			if l.Name != model.QuantileLabel && l.Name != labels.BucketLabel { // quantile and le are special labels handled below
-
+			if l.Name != model.QuantileLabel && l.Name != labels.BucketLabel {
 				lbls.WriteString(l.Name)
 				lbls.WriteString(l.Value)
+			} else if l.Name == model.QuantileLabel {
```

Comment:
Could you add some unit-tests for the specific use-case we are trying to fix here? This would help ensuring the introduced functionality.

Reply:
It's not really a fix, I just added these lines so we could obtain the value for the quantile / le at the same time we were getting the value for labels instead of doing it later.